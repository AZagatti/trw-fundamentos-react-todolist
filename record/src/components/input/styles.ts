import styled from 'styled-components'

export const Input = styled.input`
  padding: 6px 10px;
  border-radius: 4px;
  margin-right: 8px;
  border: 1px solid #333;
  &:disabled {
    background-color: #f0f0f0;
  }
`
