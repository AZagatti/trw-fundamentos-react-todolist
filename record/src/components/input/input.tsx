import { InputHTMLAttributes } from 'react'
import * as S from './styles'

export const Input = (props: InputHTMLAttributes<HTMLInputElement>) => {
  return <S.Input type="text" {...props} />
}
