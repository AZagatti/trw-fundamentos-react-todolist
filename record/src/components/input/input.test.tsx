import { render, screen } from "@testing-library/react"
import { Input } from "./input"

describe('<Input />', () => {
  it('can render correctly', () => {
    render(<Input placeholder="Input" />)
    expect(screen.getByPlaceholderText(/input/i)).toMatchSnapshot()
  })
})