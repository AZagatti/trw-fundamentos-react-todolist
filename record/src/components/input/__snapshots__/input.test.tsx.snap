// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Input /> > can render correctly 1`] = `
.c0 {
  padding: 6px 10px;
  border-radius: 4px;
  margin-right: 8px;
  border: 1px solid #333;
}

.c0:disabled {
  background-color: #f0f0f0;
}

<input
  class="c0"
  placeholder="Input"
  type="text"
/>
`;
